### PR TITLE
Fix sparser cache

### PR DESCRIPTION
**Pull request by pagreene**
Make sparser process cached results in old reading pipeline.

### CODE REVIEW

File: indra/tools/reading/read_pmids.py [hunk 1]
```diff
@@ -445,11 +445,19 @@ def get_stmts(pmids_unread, cleanup=True, sparser_version=None):
     return stmts
 
 
+def get_stmts_from_cache(pmid):
+    json_str = s3_client.get_reader_json_str('sparser', pmid)
+    stmts = []
+    if json_str is not None:
+        stmts = sparser.process_json_dict(json.loads(json_str))
```
Comment (bgyori) [inline indra/tools/reading/read_pmids.py]: I haven't tested this but this doesn't look right - wouldn't sparser.process_json_dict return a SparserProcessor, not a list of Statements? Has this worked in practice?

Comment (pagreene) [reply indra/tools/reading/read_pmids.py]: Ah, yes, it does. I misread it when I saw `sp.get_statements()`.

Comment (pagreene) [reply indra/tools/reading/read_pmids.py]: Ok...this got much harder to push the fix because of the rebase...but I have the fix ready.

**Comment by bgyori** [comment]
Great, this is working on some examples I ran

**Pull request merged**